src/prefect/experimental/workers/process.py

```diff
@@ -38,12 +46,24 @@ def _infrastructure_pid_from_process(process: anyio.abc.Process) -> str:
     return f"{hostname}:{process.pid}"
 
 
+class ProcessJobConfiguration(BaseJobConfiguration):
+    stream_output: bool = Field(template="{{ stream_output }}")
+    working_dir: Optional[Union[str, Path]] = Field(template="{{ working_dir }}")
+
+
+class ProcessVariables(BaseVariables):
+    stream_output: Optional[bool] = True
```

Comment:
We don't want `stream_output` to be `None`, so can just use `bool`. Setting a default field will cause pydantic to mark the field as optional.
```suggestion
    stream_output: bool = True
```